### PR TITLE
feat: tippy.js tooltips & interaction mode toggle

### PR DESCRIPTION
**Pull request by Dynamic6448**
- Added a new toggle input to enable/disable interaction mode
- Add tooltips to instrument buttons (interaction-mode, toggle refresh button, & auto-refresh toggle)

### CODE REVIEW

File: src/components/Input.tsx [hunk 1]
```diff
@@ -58,33 +52,26 @@ export const FileInput: React.FC<FileInputProps> = ({ options, onFileSelect, ...
...
+}
+export const ToggleInput: React.FC<ToggleInputProps> = ({ tooltip, className, ...props }) => {
+    const id = useId();
+
+    return (
+        <div>
+            <div data-tooltip-id={id} className="relative w-10 h-5">
+                <input type="checkbox" className={classNames('opacity-0 w-full h-full cursor-pointer peer', className)} {...props} />
+                <div className="absolute top-0 left-0 w-full h-full rounded-full bg-silver-700 pointer-events-none" />
+                <div className={classNames('absolute top-1 left-1 w-3 h-3 rounded-full bg-silver-600 duration-100 pointer-events-none', 'peer-checked:bg-amethyst-400 peer-checked:left-6')} />
```
Comment (MikeRomaa) [inline src/components/Input.tsx:72]: Again.
```suggestion
                <input
                    type="checkbox"
                    className={classNames('opacity-0 w-full h-full cursor-pointer peer', className)}
                    {...props}
                />
                <div className="absolute top-0 left-0 w-full h-full rounded-full bg-silver-700 pointer-events-none" />
                <div
                    className={classNames(
                        'absolute top-1 left-1 w-3 h-3 rounded-full bg-silver-600 duration-100 pointer-events-none',
                        'peer-checked:bg-amethyst-400 peer-checked:left-6',
                    )}
                />
```

File: src/components/Input.tsx [hunk 2]
```diff
@@ -58,33 +52,26 @@ export const FileInput: React.FC<FileInputProps> = ({ options, onFileSelect, ...
...
     return (
         // TODO: Replace with type="file"
-        <Input
-            readOnly
-            value={currentValue && !Array.isArray(currentValue) ? currentValue : ''}
-            className="cursor-pointer"
-            onClick={handleClick}
-            {...props}
-        />
+        <Input readOnly value={currentValue && !Array.isArray(currentValue) ? currentValue : ''} className="cursor-pointer" onClick={handleClick} {...props} />
```
Comment (MikeRomaa) [inline src/components/Input.tsx:55]: You already know.
```suggestion
        <Input
            readOnly
            value={currentValue && !Array.isArray(currentValue) ? currentValue : ''}
            className="cursor-pointer"
            onClick={handleClick}
            {...props}
        />
```

File: src/components/Input.tsx [hunk 3]
```diff
@@ -26,14 +27,7 @@ export const Input: React.FC<InputProps> = ({ label, error, className, ...props
...
                 <FiAlertCircle size={22} className="peer stroke-red-500" />
-                <span
-                    className={classNames(
-                        'absolute -top-1 left-7 px-2 py-1 bg-red-500 rounded-md opacity-0 pointer-events-none',
-                        'peer-hover:opacity-100 duration-200',
-                    )}
-                >
-                    {error}
-                </span>
+                <span className={classNames('absolute -top-1 left-7 px-2 py-1 bg-red-500 rounded-md opacity-0 pointer-events-none', 'peer-hover:opacity-100 duration-200')}>{error}</span>
```
Comment (MikeRomaa) [inline src/components/Input.tsx:30]: ...
```suggestion
                <span
                    className={classNames(
                        'absolute -top-1 left-7 px-2 py-1 bg-red-500 rounded-md opacity-0 pointer-events-none',
                        'peer-hover:opacity-100 duration-200',
                    )}
                >
                    {error}
                </span>
```

**Review by MikeRomaa — COMMENTED** [review]
I think we should use [`tippy.js`](https://www.npmjs.com/package/@tippyjs/react) instead since it has a handy prop called `appendTo` that lets us portal the tooltip out of the workspace. That way, the tooltip is not affected by the zoom level set by the workspace transform.

It also foregoes the `data-tooltip-id` props and instead renders based on children. For example:
```tsx
<Tippy content="Refresh Instrument" appendTo={document.body}>
    <button className="ml-auto" onClick={refresh}>
        <HiRefresh className="cursor-pointer active:text-silver-500" size={22} />
    </button>
</Tippy>
```

Let me know what you think, always open to discussing alternatives.

File: src/components/Input.tsx [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import React, { InputHTMLAttributes, useCallback, useState } from 'react';
 import { FiAlertCircle } from 'react-icons/fi';
+import Tippy from '@tippyjs/react';
 import classNames from 'classnames';
 import { open, OpenDialogOptions } from '@tauri-apps/api/dialog';
 import Slider, { SliderProps } from 'rc-slider';
 import 'rc-slider/assets/index.css';
+import 'tippy.js/dist/tippy.css';
```
[the remark this answers is not in the document]
Comment (MikeRomaa) [reply src/components/Input.tsx:8]: +1 for importing at the root

File: src/components/Input.tsx [hunk 5]
```diff
@@ -68,23 +70,26 @@ export const FileInput: React.FC<FileInputProps> = ({ options, onFileSelect, ...
...
+            />
+            <div className="absolute top-0 left-0 w-full h-full rounded-full bg-silver-700 pointer-events-none" />
+            <div
+                className={classNames(
+                    'absolute top-1 left-1 w-3 h-3 rounded-full bg-silver-600 duration-100 pointer-events-none',
+                    'peer-checked:bg-amethyst-400 peer-checked:left-6',
+                )}
+            />
+        </div>
+    </Tippy>
```
Comment (MikeRomaa) [inline src/components/Input.tsx:94]: Let's keep it simple and just define Tooltips around the component when necessary. I'd just leave the `ToggleInput` as it was and throw the Tippy stuff into `instrument.tsx` directly.

File: src/components/Instrument.tsx [hunk 6]
```diff
@@ -150,15 +165,26 @@ export const Instrument: React.FC<Element> = ({ uuid, name, element, x, y, width
...
             </div>
             <div className="absolute w-full h-full box-content border-2 border-silver-700 bg-black">
                 <InstrumentFrame ref={iframeRef} name={name} width={width} height={height} />
             </div>
+            <div
+                className={classNames(
+                    'absolute w-full h-full box-content border-2 border-silver-700 bg-transparent',
+                    { 'pointer-events-none': interactable },
+                )}
+            />
```
Comment (MikeRomaa) [inline src/components/Instrument.tsx:187]: Wait I just noticed this, what's this for? Can't we can just toggle `pointer-events-none` directly on the container (lines 145 - 156)?

File: package.json [hunk 7]
```diff
@@ -28,6 +29,7 @@
     "react-redux": "^8.1.1",
     "react-router": "^6.14.1",
     "react-zoom-pan-pinch": "^3.1.0",
+    "tippy.js": "^6.3.7",
```
Comment (MikeRomaa) [inline package.json:32]: You should only need the react package.

**Review by MikeRomaa — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (Dynamic6448) [reply package.json:32]: I need the base package to import styles

File: src/components/Instrument.tsx (hunk 6 above)
Comment (Dynamic6448) [reply src/components/Instrument.tsx:187]: Nope, then you won't be able to click any buttons on the instrument. Putting it at line 179 works

File: src/components/Input.tsx (hunk 5 above)
Comment (Dynamic6448) [reply src/components/Input.tsx:94]: This doesn't work for some reason
```tsx
<Tippy content="Toggle Auto-Refresh" appendTo={document.body}>
    <ToggleInput onClick={handleWatch} />
</Tippy>
```

Comment (MikeRomaa) [reply src/components/Input.tsx:94]: Might need to forward ref on the toggle input component.

File: src/components/Instrument.tsx (hunk 6 above)
Comment (MikeRomaa) [reply src/components/Instrument.tsx:187]: I could've sworn I had that working before. I'll take a closer look when I get home.

**Pull request merged**